sharktank/tests/models/llama/benchmark-tests.py

```diff
@@ -0,0 +1,310 @@
+
+    def iree_compile(
+        self, mlir_path: str, output_file: str, args: List[str], cwd: str | Path
+    ):
+        """Compiles an input MLIR file to an output .vmfb file.
+        This assumes that the `iree-compile` command is available (usually via PATH).
+        Args:
+            mlir_path: Path to the input MLIR file.
+            output_file: Path for the output file. The directory must already exist.
+            args: List of arguments to pass to `iree-compile`.
+            cwd: current working directory
+        Raises Exception if compilation fails for some reason.
+        """
+        cmd = self.get_compile_cmd(mlir_path, output_file, args)
+        logging.getLogger().info(f"Launching compile command:\n" f"cd {cwd} && {cmd}")
+        proc = subprocess.run(cmd, shell=True, capture_output=True, cwd=cwd)
+        return_code = proc.returncode
+        if return_code != 0:
+            raise Exception(f"{cmd} failed to compile.")
+
+    def iree_benchmark_module(
+        self,
+        hip_device_id: str,
+        vmfb_name: str,
+        irpa_path: str,
+        args: List[str],
+        cwd: str | Path,
+    ):
+        """Runs a compiled program with the given args using `iree-benchmark-module`.
+        This assumes that the `iree-benchmark-module` command is available (usually via PATH).
+        Args:
+            vmfb_name: Name of the .vmfb file (relative to `cwd`).
+            args: List of arguments to pass to `iree-benchmark-module`.
+            cwd: Working directory to run the command within. (either string or Path works)
+            compile_cmd: Command used to compile the program, for inclusion in error messages.
+        Raises Exception if running fails for some reason.
+        """
+        benchmark_args = [
+            f"ROCR_VISIBLE_DEVICES={hip_device_id}",
+            "iree-benchmark-module",
+            f"--device=hip://{hip_device_id}",
+            "--hip_use_streams=true",
+            "--hip_allow_inline_execution=true",
+            "--device_allocator=caching",
+            f"--module={vmfb_name}",
+            f"--parameters=model={irpa_path}",
+        ]
+        benchmark_args += args
+        cmd = subprocess.list2cmdline(benchmark_args)
+        logging.getLogger().info(f"Launching run command:\n" f"cd {cwd} && {cmd}")
+        proc = subprocess.run(cmd, shell=True, capture_output=True, cwd=cwd)
+        return_code = proc.returncode
+        if return_code != 0:
+            raise Exception(f"{cmd} failed to run")
+
+
+class BenchmarkLlama3_1_8B_f16(BaseBenchmarkTest):
+    def setUp(self):
+        # TODO: add numpy files to Azure and download from it
+        self.repo_root = os.getenv("SHARK_PLATFORM_REPO_ROOT")
+        artifacts_dir = "/data/extra/models/llama3.1_8B/"
+        self.irpa_path = artifacts_dir + "llama8b_f16.irpa"
+        self.irpa_path_fp8 = artifacts_dir + "llama8b_fp8.irpa"
+        self.output_mlir = self.repo_root + "llama8b_f16.mlir"
```

Comment:
output files should be temporary. We don't want them lasting or being picked up accidentally on subsequent runs

Reply:
Added a function to remove the output files at the end of each test